README.md

```diff
@@ -1343,18 +1344,145 @@ We run task **3** the output would be:
 * [Global](#usage-env-global)
 
+<a name="env-declaration"></a>
+#### Declaration
+
+There are multiple ways to declare environmental variables, all of which are suited for specific suitcases.
+
+##### Simple
+
+The most ordinary one is the definition of a simple `KEY=Value` pair, which is reminiscent of tools like [dotenv](https://www.npmjs.com/package/dotenv) and [bash scripts](https://www.gnu.org/software/bash/). Values can use other variables as values, which are interpolated at runtime, using the `${variable}` syntax.
+
+> **Note:** The `$variable` syntax is currently not supported!
+
+> **Note:** Extended interpolation like `${variable:-default}` are currently unsupported!
+
+```toml
+RUST_BACKTRACE = 1
+BOOL_VALUE = true
+COMPOSITE = "${BOOL_VALUE} ${RUST_BACKTRACE}"
+```
+
+##### List
+
+`cargo-make` also supports lists, which are joined using `;` at runtime.
+
+```toml
+LIST_VALUE = [ "VALUE1", "VALUE2", "VALUE3" ]
+```
+
+##### Script
+
+`cargo-make` supports the use of simple scripts. The output of the said script will then determine the value of the environmental variable.
+
+The script's object has two additional arguments: `multiline` and `depends_on`. If `multiple` is set to `true`, the supplied script will be evaluated as a script with multiple lines. `depends_on` is a list of environmental variables this script depends on, which is taken into account during reordering if unset `cargo-make` will try to guess the variables used during reordering.
+
+> **Note:** This uses `sh` for script evaluation ([spec](https://pubs.opengroup.org/onlinepubs/9699919799/)), other runners like `duckscript`, `rust`, etc. are **not** supported.
+
+```toml
+EVALUATED_VAR = { script = ["echo SOME VALUE"] }
+```
+
+##### Decode Map
+
+`cargo-make` supports the use of mappings where a `source` is matched against a dictionary of possible `mapping`s, where each key of the `mapping` is compared against the evaluated `source` value. Should the key and `source` be the same, the corresponding value to the key will be the value of the environmental variable. If no key is matched, the `default_value` is used if provided. Otherwise, it will default to an empty string instead.
+
+```toml
+LIBRARY_EXTENSION = { source = "${CARGO_MAKE_RUST_TARGET_OS}", default_value = "unknown", mapping = {"linux" = "so", "macos" = "dylib", "windows" = "dll", "openbsd" = "so" } }
+```
+
+##### Path
+
+`cargo-make` supports the use of glob syntax to find all files and directories in a given directory. The list of files will be joined using `;` during execution.
+
+```toml
+PATH_GLOB = { glob = "./src/**/mod.rs", include_files = true, include_dirs = false, ignore_type = "git" }
+```
+
+##### Conditional
+
+`cargo-make` supports conditional variables, which are set to the `value` specified if the `condition` evaluates to true. To learn more about conditions, refer to [this chapter]
+
+##### Unset
+
+Variables can be unset, which will unset their value and reset it to an empty string.
```

Comment:
Let's simplify here since its not setting it to empty string. its removing it.
so maybe just write: "Variables can be unset." as i think its enough